File: android/app/src/main/kotlin/com/example/access_checkout_flutter_demo/AccessCheckoutViewFactory.kt
import android.content.Context
import androidx.lifecycle.LifecycleOwner
import com.example.access_checkout_flutter_demo.AccessCheckoutView
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class AccessCheckoutViewFactory(
    private val messenger: BinaryMessenger,
    private val channel: String,
    private val lifecycleOwner: LifecycleOwner
) :
    PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        val methodChannel = MethodChannel(messenger, channel)
        val creationParams: Map<String, Any> = args as Map<String, Any>
        return AccessCheckoutView(lifecycleOwner, methodChannel, context, creationParams)
    }
}

File: access_checkout_flutter_native_sdk_demo/ios/Runner/AccessCheckoutViewFactory.swift
import Flutter
import UIKit

class AccessCheckoutViewFactory: NSObject, FlutterPlatformViewFactory {
    private var messenger: FlutterBinaryMessenger
    private var channel: String 

    init(messenger: FlutterBinaryMessenger, channel: String) {
        self.messenger = messenger
        self.channel = channel
        super.init()
    }

    
    func create(
        withFrame frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) -> FlutterPlatformView {
        let creationParams = args as? [String: Any] ?? [:]
        return AccessCheckoutView(
            methodChannel: FlutterMethodChannel(name: channel, binaryMessenger: messenger),
            frame: frame,
            viewIdentifier: viewId,
            binaryMessenger: messenger,
            creationParams: creationParams
        )
    }
    
    /// Implementing this method is only necessary when the `arguments` in `createWithFrame` is not `nil`.
    public func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        return FlutterStandardMessageCodec.sharedInstance()
    }
}
